File: energy_frontend/src/app/DTO/ChatMessageWebSocketDTO.ts
export class ChatMessageWebSocketDTO {
    type : string = "";
    message : string = "";
    senderId : number = 0;

    constructor(type : string, message : string, receiverId : number, senderId : number) {
        this.type = type;
        this.message = message;
        this.senderId = senderId;
    }
}
  

File: microservices/energychat/src/main/java/com/cezar/energychat/DTO/ChatMessageWebSocketDTO.java
package com.cezar.energychat.DTO;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
public class ChatMessageWebSocketDTO {
    String type; // = MESSAGE | SEEN | TYPING
    String message = "";
    long senderId;
}
